### ISSUE TITLE
Building a container with a huge amount of standard output crashes the build

### ISSUE THREAD

**Issue by drubin** [garden-io/garden#399]
Because we proxy the output of `docker build` and display it on failure if a container produces a huge amount of logs we can't handle it. 

```
Failed building shipping. Here is the output:
————————————————————————————————————————————————————————————————————————————————
stdout maxBuffer exceeded
————————————————————————————————————————————————————————————————————————————————
```

We need to figure out how we want to handle logs from commands. (kubectl and others also have the same problems.

We should figure out an abstract solution to handling this type of sinario when the stream output is unpredictable but we should handle it in the same way regardless of the setup.

Possible options are

1) store temp buffer of recent logs
2) write raw logs to file
3) Do some combo of showing the buffer and the link to the raw logs

**Comment by edvald** [comment]
I suggest we do the following, either together or separately. The first fixes the bug, the others are more an enhancement:
1. Wrap `execa` to only keep the last X lines and last Y bytes of logs in memory, instead of just buffering the whole thing. This also serves to reduce spam in the console.
2. In the same wrapper, emit each written line as a log event that can be routed to UIs or elsewhere.
3. Pipe the logs to some form of storage. Could be just files to begin with, but no matter what we need to have automated cleanup in place.

**Comment by edvald** [comment]
Fixed in #602
